app/src/main/java/com/task/ui/component/details/DetailsViewModel.kt

```diff
@@ -9,14 +9,15 @@ import com.task.data.Resource
 import com.task.data.dto.recipes.RecipesItem
 import com.task.ui.base.BaseViewModel
 import com.task.utils.wrapEspressoIdlingResource
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 /**
  * Created by AhmedEltaher
  */
-
+@HiltViewModel
 open class DetailsViewModel @Inject constructor(private val dataRepository: DataRepositorySource) : BaseViewModel() {
```

Comment:
https://developer.android.com/reference/androidx/hilt/lifecycle/ViewModelInject

Reply:
Did you read the link you posted?!

Reply:
it is deprecated already :joy:
sorry man my mistake